st2actions/st2actions/runners/ssh/paramiko_ssh.py

```diff
@@ -483,5 +485,14 @@ def _get_pkey_object(self, key):
             else:
                 return key
 
-        msg = 'Invalid or unsupported key type'
+        # If a user passes in something which looks like file path we throw a more friendly
+        # exception letting the user know we expect the contents a not a path.
+        # Note: We do it here and not up the stack to avoid false positives.
+        contains_header = PRIVATE_KEY_HEADER in key.lower()
+        if not contains_header and (key.count('/') >= 1 or key.count('\\') >= 1):
```

Comment:
As noted in the comment - the best place to check for this would be in the runner, but doing there could result in false positives so I do it here after we try to load the key.